rustyms-py/src/lib.rs

```diff
@@ -0,0 +1,1262 @@
+        match &self.0.c_term {
+            Some(m) => Some(Modification(m.clone())),
+            None => None,
+        }
+    }
+
+    /// Sequence of the peptide including modifications.
+    ///
+    /// Returns
+    /// -------
+    /// list[SequenceElement]
+    ///
+    #[getter]
+    fn sequence(&self) -> Vec<SequenceElement> {
+        self.0
+            .sequence
+            .iter()
+            .map(|x| SequenceElement(x.clone()))
+            .collect()
+    }
+
+    /// For each ambiguous modification list all possible positions it can be placed on. Indexed by the ambiguous modification id.
+    ///
+    /// Returns
+    /// -------
+    /// list[list[int]]
+    ///
+    #[getter]
+    fn ambiguous_modifications(&self) -> Vec<Vec<usize>> {
+        self.0.ambiguous_modifications.clone()
+    }
+
+    /// Stripped sequence, meaning the sequence without any modifications.
+    ///
+    /// Returns
+    /// -------
+    /// str
+    ///
+    #[getter]
+    fn stripped_sequence(&self) -> String {
+        self.0.sequence.iter().map(|x| x.aminoacid.char()).collect()
+    }
+
+    // TODO: How to get the charge as an integer?
+    // /// The adduct ions, if specified.
+    // #[getter]
+    // fn charge_carriers(&self) -> Vec<> {
+    //     todo!()
+    // }
+
+    /// Get a copy of the peptide with its sequence reversed.
+    ///
+    /// Returns
+    /// -------
+    /// LinearPeptide
+    ///
+    fn reverse(&self) -> LinearPeptide {
+        LinearPeptide(self.0.reverse())
+    }
+
+    /// Gives the formula for the whole peptide. With the global isotope modifications applied.
+    ///
+    /// Returns
+    /// -------
+    /// MolecularFormula | None
+    ///
+    fn formula(&self) -> Option<MolecularFormula> {
+        match self.0.formula() {
+            Some(f) => Some(MolecularFormula(f)),
+            None => None,
+        }
+    }
+
+    /// Generate the theoretical fragments for this peptide, with the given maximal charge of the fragments, and the given model. With the global isotope modifications applied.
+    ///
+    /// Parameters
+    /// ----------
+    /// max_charge : int
+    ///    The maximal charge of the fragments.
+    /// model : str
+    ///   The model to use for the fragmentation. One of: ``all``, ``cid_hcd``, ``etcid``, ``etd``, ``ethcd``
+    ///
+    /// Returns
+    /// -------
+    /// list[Fragment]
+    ///   The theoretical fragments.
+    ///
+    fn generate_theoretical_fragments(
+        &self,
+        max_charge: i16,
+        model: &str,
+        // peptide_index: usize, TODO: Required for linear peptide?
+    ) -> PyResult<Vec<Fragment>> {
+        let model = match_model(model)?;
+        Ok(self
+            .0
+            .generate_theoretical_fragments(
+                rustyms::system::Charge::new::<rustyms::system::e>(max_charge as f64),
+                &model,
+                0, // TODO: Don't hard code?
+            )
+            .unwrap()
```

Comment:
At the rust side this function should not be able to crash anymore (now that I build support for B/Z).

Reply:
Should I replace the `unwrap` with something else?

Reply:
No the result of this function is just a list of fragments it is not a `Result` anymore, so Rust should just give a compilation error.